pkg/proofs/ladder.go

```diff
@@ -4,16 +4,16 @@ func FullBinaryLadderSteps(target uint32) (r []uint32) {
 	r = make([]uint32, 0)
 	var i uint32 = 0
 	for i < target {
-		r = append(r, i)
+		r = append( /*@ perm(1/2), @*/ r, i)
 		i = i << 1
```

Comment:
This is an infinite loop. `i` starts at 0, and `0<<1 = 0`.

Reply:
Opened issue #6 to track this

Reply:
Fixed in b0e0d4a9be05e66518a72c45ddceb884f949d057 ! Thanks for catching!